### PR TITLE
Deploy odh google spark operator to zero cluster.

### PR DESCRIPTION
**Pull request by HumairAK**
Related: https://github.com/operate-first/support/issues/241

### CODE REVIEW

**Comment by HumairAK** [comment]
currently using @guimou 's repo, but will switch to odh-manifests when https://github.com/opendatahub-io/odh-manifests/pull/427 is merged

**Comment by HumairAK** [comment]
the readme seems to suggest there may be a need for an additional SA / rbac, but the operator has clusterwide privileges for pods so I'm not sure why that would be necessary...

File: kfdefs/base/google-spark-operator/kfdef.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  name: opendatahub
+spec:
+  applications:
+    - kustomizeConfig:
+        repoRef:
+          name: guimou
+          path: google-spark-operator/operator
+      name: google-spark-operator
+  repos:
+    - name: manifests
```
Comment (4n4nd) [inline kfdefs/base/google-spark-operator/kfdef.yaml:16]: I don't think we need the "manifests" repo here since it is not being used in the kfdef

Comment (HumairAK) [reply kfdefs/base/google-spark-operator/kfdef.yaml:16]: the eventual goal is to switch to this repo

Comment (tumido) [reply kfdefs/base/google-spark-operator/kfdef.yaml:16]: I don;t mid either way. The fact is that keeping it in here makes this manifest eligible for ODH z-stream update PRs  which makes us keep an eye on this manifest more closely. The fact that there's an open PR for this code makes me assume that in 1 or 2 z-stream releases we will be able to switch over to ODH upstream repo here.

**Review by tumido — COMMENTED** [review]
/approve

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Review by 4n4nd — APPROVED** [review]
/lgtm

**Pull request merged**